### PR TITLE
Add TLS Certificate Pinning

### PR DESCRIPTION
**Pull request by jkoenig134**

### CODE REVIEW

File: _docs_operate/configuration.md [hunk 1]
```diff
@@ -110,6 +112,56 @@ You can validate the config using our [schema file](https://raw.githubuserconten
 
 The debug flag configures if the Connector is set to **production** or **debug** mode. Defaults to `false`. Can also be configured using the environment variable `DEBUG`.
 
+### enforceCertificatePinning `available since version 6.5.0` {#enforceCertificatePinning}
+
+The `enforceCertificatePinning` flag configures if the Connector should enforce certificate pinning. Defaults to `false`.
```
Comment (britsta) [inline _docs_operate/configuration.md:117]: The `enforceCertificatePinning` flag configures ~if~whether the Connector should enforce certificate pinning.

File: _docs_operate/configuration.md [hunk 2]
```diff
@@ -110,6 +112,56 @@ You can validate the config using our [schema file](https://raw.githubuserconten
...
+
+If enabled the Connector will only accept TLS certificates that match the SHA256 fingerprints specified in the `pinnedTLSCertificateSHA256Fingerprints` object. Hostname not configured at all can not be accessed by the connector anymore.
```
Comment (britsta) [inline _docs_operate/configuration.md:119]: - Insert comma after "If enabled" please.
- Write Connector capitalized in the second sentence.
- Also, the beginning of the second sentence seems a little incomplete.
- Detail: "cannot" is much more often used by us in comparison with "can not".

File: _docs_operate/configuration.md [hunk 3]
```diff
@@ -110,6 +112,56 @@ You can validate the config using our [schema file](https://raw.githubuserconten
...
+```text
+Connecting to <ip>
+...
+DONE
+sha256 Fingerprint=AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA
+```
+
+You can simply copy the fingerprint after `sha256 Fingerprint=` and use it.
+
+If you use another way to acquire the fingerprint the connector understands multiple formats like `AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA`, `AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`, `aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa` and `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`.
```
Comment (britsta) [inline _docs_operate/configuration.md:144]: - Insert comma after "If you use another way to acquire the fingerprint".
- Write Connector capitalized please.

File: .vscode/tasks.json [hunk 4]
```diff
@@ -9,6 +9,7 @@
                 "kind": "test",
                 "isDefault": true
             },
+            "options": { "shell": { "args": [] } },
```
Comment (Milena-Czierlinski) [inline .vscode/tasks.json:12]: Why did you add this?

File: _docs_operate/configuration.md [hunk 5]
```diff
@@ -110,6 +112,56 @@ You can validate the config using our [schema file](https://raw.githubuserconten
...
+```text
+Connecting to <ip>
+...
+DONE
+sha256 Fingerprint=AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA
+```
+
+You can simply copy the fingerprint after `sha256 Fingerprint=` and use it.
+
+If you use another way to acquire the fingerprint, the Connector understands multiple formats like `AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA`, `AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`, `aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa` and `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`.
```
Comment (britsta) [inline _docs_operate/configuration.md:144]: The two lines `AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA` and `aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa:aa` are so long that they extend beyond the boundaries of the text area. And when scrolling, the TOC starts to covering the lines. If no correctness is lost, please make them a little shorter.

Comment (jkoenig134) [reply _docs_operate/configuration.md:144]: I can not change the length. `AA:AA` is not a correct value.

File: _docs_operate/configuration.md [hunk 6]
```diff
@@ -110,6 +112,56 @@ You can validate the config using our [schema file](https://raw.githubuserconten
...
+
+**Getting the SHA256 fingerprint of a certificate:**
+
+```bash
+echo -n | openssl s_client -connect <hostname>:443 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > cert.pem
+openssl x509 -noout -in cert.pem -fingerprint -sha256
+rm cert.pem
+```
+
+this will output something similar to:
```
Comment (Milena-Czierlinski) [inline _docs_operate/configuration.md:133]: Please capitalize the beginning of the sentence.

File: _docs_operate/configuration.md (hunk 5 above)
Comment (britsta) [reply _docs_operate/configuration.md:144]: I see. As it is breaking our page the way it is right now, I see two alternatives to prevent this:

- Use `<code>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA:<wbr>AA</code>` as an alternative for flexible line break, altough its ugly in the code.
- Or put them in a separate code area (with scroll bar like we have in our json blocks)

File: _docs_operate/configuration.md (hunk 6 above)
Comment (jkoenig134) [reply _docs_operate/configuration.md:133]: this is in the middle of a sentence

File: .vscode/tasks.json (hunk 4 above)
Comment (jkoenig134) [reply .vscode/tasks.json:12]: only way I can start jekyll on mac

File: _docs_operate/configuration.md (hunk 6 above)
Comment (Milena-Czierlinski) [reply _docs_operate/configuration.md]: > this is in the middle of a sentence

Debatable at least. 😅

**Review by Milena-Czierlinski — DISMISSED** [review]

**Review by britsta — APPROVED** [review]

File: _docs_operate/security-considerations.md [hunk 7]
```diff
@@ -79,6 +79,12 @@ The Connector does need to access its database. Access to other networks or syst
 
 Depending on the integration setup, access to the Connector from the internal network could be blocked for the majority of requests. Usually, only requests from the integration systems, the developers or administrators need to be allowed.
 
+### TLS Certificate Pinning
+
+To make the Connector resistant against man-in-the-middle attacks, it is recommended to use [TLS certificate pinning](https://www.ssl.com/blogs/what-is-certificate-pinning/). This means that the Connector only accepts a specific certificate for the connection to the Backbone. This certificate [should be pinned in the Connector configuration]({% link _docs_operate/configuration.md %}#pinnedTLSCertificateSHA256Fingerprints).
+
+As an additional security mechanism, we recommend to [enforce certificate pinning]({% link _docs_operate/configuration.md %}#enforceCertificatePinning) to make sure that the Connector does not access any other HTTPS domain other than the ones specified in the configuration.
```
Comment (stnmtz) [inline _docs_operate/security-considerations.md:86]: Warning box: Certificate pinning adds additional effort to the Connector administrators, as for every hostname and TLS certificate, the fingerprints must be regularly updated over the config. Be aware that the Connector might reject TLS connections and cease to function if this configuration is not maintained.
Be also aware that the Connector might reject TLS connections seemingly at random, if on the server infrastructure, multiple TLS certificates or redirection is used.
As TLS certificates are maintained by the domain owner of the hostname, establishing a communication channel between Connector administrator and TLS server administrator is recommended.

File: _docs_operate/configuration.md [hunk 8]
```diff
@@ -110,6 +112,63 @@ You can validate the config using our [schema file](https://raw.githubuserconten
 
 The debug flag configures if the Connector is set to **production** or **debug** mode. Defaults to `false`. Can also be configured using the environment variable `DEBUG`.
 
+### enforceCertificatePinning `available since version 6.5.0` {#enforceCertificatePinning}
+
+The `enforceCertificatePinning` flag configures whether the Connector should enforce certificate pinning. Defaults to `false`.
+
+If enabled, the Connector will only accept TLS certificates that match the SHA256 fingerprints specified in the `pinnedTLSCertificateSHA256Fingerprints` object. If a Hostname is not configured at all, it cannot be accessed by the Connector anymore.
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply _docs_operate/configuration.md:119]: when example.com is not configured every call to `https://example.com` will fail.

File: _docs_operate/configuration.md [hunk 9]
```diff
@@ -110,6 +112,63 @@ You can validate the config using our [schema file](https://raw.githubuserconten
...
+
+### pinnedTLSCertificateSHA256Fingerprints `available since version 6.5.0` {#pinnedTLSCertificateSHA256Fingerprints}
+
+The `pinnedTLSCertificateSHA256Fingerprints` object contains the SHA256 fingerprints of the TLS certificates that the Connector should accept. The fingerprints must be in the format `SHA256:<fingerprint>`. The Connector will only accept TLS certificates that match the fingerprints specified in this object.
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply _docs_operate/configuration.md:123]: nah, we mention pilot nowhere.

**Review by stnmtz — APPROVED** [review]
